### ISSUE TITLE
Inserting multiple data items can trigger failure loop on sqlite

### ISSUE THREAD

**Issue by awrichar** [hyperledger/firefly#1196]
When `InsertDataArray()` is called on a database that does _not_ support `MultiRowInsert` (such as sqlite), the code branch that is triggered specifically passes `false` for `requestConflictEmptyResult` (meaning that any row which fails to insert due to a key conflict should trigger a failure and rollback of the database transaction):

https://github.com/hyperledger/firefly/blob/091bc226914033bf7c173df30368e5b77128af04/internal/database/sqlcommon/data_sql.go#L215

However, the way that this method is used during batch persistence implies that it should be a graceful failure and should allow the caller to fall back to individual upsert operations, in case some data items are old and some are new:

https://github.com/hyperledger/firefly/blob/091bc226914033bf7c173df30368e5b77128af04/internal/events/persist_batch.go#L268-L271

In other words, it seems like the code in sqlcommon is wrong, and should use `requestConflictEmptyResult = true`.

**Comment by awrichar** [comment]
#1198 fixes the root issue here (the fact that the transaction is aborted when the optimized query fails, instead of allowing a graceful fallback)

#1197 actually addresses the related issue that I've filed separately as #1199 (the fact that Postgres databases are currently misconfigured)
